main.tf

```diff
@@ -207,12 +207,25 @@ resource aws_vpn_gateway_route_propagation "vpn_routes_private" {
 # Flow Logs
 ###########
 
+resource aws_flow_log "vpc_log" {
+  count                = "${var.build_s3_flow_logs ? 1 : 0}"
+  log_destination      = "${aws_s3_bucket.vpc_log_bucket.arn}"
+  log_destination_type = "s3"
+  vpc_id               = "${aws_vpc.vpc.id}"
+  traffic_type         = "ALL"
+}
+
+resource "aws_s3_bucket" "vpc_log_bucket" {
+  count = "${var.build_s3_flow_logs ? 1 : 0}"
+  tags  = "${merge(local.base_tags, map("Name", format("%s-VPNGateway", var.vpc_name), "transitvpc:spoke", var.spoke_vpc), var.custom_tags)}"
```

Comment:
We should add the appropriate variables to allow for using an existing bucket, setting bucket name, setting encryption on this bucket.  We should probably also consider adding a life cycle rule to purge old records and a user defined age.

Reply:
We should update the value of the `Name` tag, as it still references the VPN Gateway resource.  I would also recommend removal of the `transitvpc:spoke` tag from the S3 bucket, as it is only used by lambda functions to dynamically find and configure VPC Gateways\VPN Connections.  It might confuse things or people if found on other resources.

Reply:
Thank , I have made an update to add extra option for the bucket.
tag for the bucket is updated.